### PR TITLE
Add a new parameter to control how time to be formatted on screen

### LINKED ISSUE #24813: Need to be able to display times on console in DTIME
## Reason
For depletion calculations in reactor analysis, the time step sizes are sometimes in days and the end time can reach few years.  Printing time in seconds makes the screen output non-intuitive. It is desired to print times in DTIME format (https://www.gnu.org/software/pspp/manual/html_node/Time-and-Date-Formats.html) if users want.

## Design
I propose to add a parameter in `Console` like `time_format` that users can change how times and time step sizes look. The default can be in second without the unit. It is noted that `Console` already had two parameters `time_precision` and `scientific_time` to control how times are outputted. Adding a new parameter makes sense to me. It is also desired that `Console` provides a setter so that applications can change the format programmatically.

## Impact
It is not going to have any impact to calculations but rather a new feature that improves the screen outputs.

### PR DESCRIPTION
**Pull request by YaqiWang**
To close #24813.

### CODE REVIEW

**Comment by YaqiWang** [comment]
With this change, the screen print-out of one of our calculations changed from something like
```
Time Step 7, time = 3.456e+06, dt = 1.728e+06
```
to
```
Time Step 7, time = 40d , dt = 20d 
```

File: framework/src/outputs/Console.C [hunk 1]
```diff
@@ -455,6 +453,49 @@ Console::writeTimestepInformation(bool output_dt)
...
+  std::ostringstream oss;
+  if (_time_format == "plain" || _time_format == "second")
+  {
+    if (_precision > 0)
+      oss << std::setw(_precision) << std::setprecision(_precision) << std::setfill('0')
+          << std::showpoint;
+    if (_scientific_time)
+      oss << std::scientific;
+    oss << t;
+    if (_time_format == 1)
```
Comment (GiudGiud) [inline framework/src/outputs/Console.C:468]: use enum

Comment (YaqiWang) [reply framework/src/outputs/Console.C:468]: Yeah, I was using `unsigned int _time_format` initially, but that makes the setter interface harder.

File: framework/src/outputs/Console.C [hunk 2]
```diff
@@ -455,6 +453,49 @@ Console::writeTimestepInformation(bool output_dt)
...
+  if (_time_format == "plain" || _time_format == "second")
+  {
+    if (_precision > 0)
+      oss << std::setw(_precision) << std::setprecision(_precision) << std::setfill('0')
+          << std::showpoint;
+    if (_scientific_time)
+      oss << std::scientific;
+    oss << t;
+    if (_time_format == "second")
+      oss << "s";
```
Comment (GiudGiud) [inline framework/src/outputs/Console.C:469]: should there be a space here?

File: framework/src/outputs/Console.C [hunk 3]
```diff
@@ -66,6 +66,11 @@ Console::validParams()
   params.addParam<unsigned int>(
       "time_precision",
       "The number of significant digits that are printed on time related outputs");
+  MooseEnum time_format("plain=0 second=1 dtime=2", "plain");
+  params.addParam<MooseEnum>(
+      "time_format",
+      time_format,
+      "The format for the printed times ('dtime' means a format like 1d 01:01:0.1s)");
```
Comment (GiudGiud) [inline framework/src/outputs/Console.C:73]: add to time formatting parameter group

File: framework/src/outputs/Console.C [hunk 4]
```diff
@@ -455,6 +453,49 @@ Console::writeTimestepInformation(bool output_dt)
...
+          << mins << ":";
+
+      if (_precision > 0)
+        oss << std::setw(_precision) << std::setprecision(_precision) << std::setfill('0')
+            << std::showpoint;
+      if (_scientific_time)
+        oss << std::scientific;
+      else if (second < 10)
+        oss << "0";
+      oss << second;
```
Comment (GiudGiud) [inline framework/src/outputs/Console.C:493]: needs a test in MOOSE

**Review by GiudGiud — COMMENTED** [review]
this is a good idea

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the contribution!

**Pull request merged**